learn/dynamic-application-configuration/waypoint.hcl

```diff
@@ -0,0 +1,75 @@
+project = "learn-dynamic-go"
+
+app "dynamic-go" {
+  labels = {
+    "service" = "dynamic-go",
+  }
+
+  config {
+    env = {
+      "USERNAME" = dynamic("vault", {
+        path = "database/creds/readonly"
+        key  = "username"
+      })
+      "PASSWORD" = dynamic("vault", {
+        path = "database/creds/readonly"
+        key  = "password"
+      })
+
+      HOST   = var.postgres_ip
+      PORT   = var.postgres_port
+      DBNAME = var.postgres_dbname
+    }
+  }
+
+  build {
+    use "pack" {}
+    registry {
+      use "docker" {
+        image = "devopspaladin/hello-database"
```

Comment:
Should this be more generic and or turned into an input variable?

Reply:
I agree! This popped in my head once or twice while I was making this example but I never revisited it. 😄

Reply:
Lol, I already had vars `image_name` and `image_tag` in the Waypoint file, but forgot to use them here! 😅